deepchecks/nlp/checks/data_integrity/text_property_outliers.py

```diff
@@ -79,99 +79,118 @@ def run_logic(self, context: Context, dataset_kind: DatasetKind) -> CheckResult:
+            # Sort the result map based on the length of indices and if there are any error message associated to
+            # any property, keep that property at the very end.
+            sorted_result_items = sorted(result.items(),
+                                         key=lambda x: len(x[1].get('indices', [])) if isinstance(x[1], dict) else 0,
+                                         reverse=True)
 
             for property_name, info in sorted_result_items:
-                # If info is string it means there was error
-                if isinstance(info, str):
-                    no_outliers = pd.concat([no_outliers, pd.Series(property_name, index=[info])])
-                elif len(info['indices']) == 0:
-                    no_outliers = pd.concat([no_outliers, pd.Series(property_name, index=['No outliers found.'])])
-                else:
-                    if len(display) < self.n_show_top:
-                        dist = df_properties[property_name]
-                        if len(dist[~pd.isnull(dist)]) >= self.min_samples:
-                            lower_limit = info['lower_limit']
-                            upper_limit = info['upper_limit']
-
-                            fig = get_text_outliers_graph(
-                                dist=dist,
-                                data=dataset.text,
-                                lower_limit=lower_limit,
-                                upper_limit=upper_limit,
-                                dist_name=property_name,
-                                is_categorical=property_name in cat_properties
-                            )
-
-                            display.append(fig)
-                        else:
-                            no_outliers = pd.concat(
-                                [no_outliers, pd.Series(property_name, index=[
-                                    f'Not enough non-null samples to compute'
-                                    f' properties (min_samples={self.min_samples}).'
-                                ])]
-                            )
+                try:
+                    # If info is string it means there was error
+                    if isinstance(info, str):
+                        no_outliers = pd.concat([no_outliers, pd.Series(property_name, index=[info])])
+                    elif len(info['indices']) == 0:
+                        no_outliers = pd.concat([no_outliers, pd.Series(property_name, index=['No outliers found.'])])
                     else:
-                        no_outliers = pd.concat([no_outliers, pd.Series(property_name, index=[
-                            f'Outliers found but not shown in graphs (n_show_top={self.n_show_top}).'])])
+                        if len(display) < self.n_show_top:
+                            dist = df_properties[property_name]
+                            if len(dist[~pd.isnull(dist)]) >= self.min_samples:
```

Comment:
maybe move this if into the first loop - if not enough samples raise relevant exception